.github/workflows/ci_workflows.yml

```
@@ -0,0 +1,123 @@
+name: CI
+
+on:
+  push:
+    branches:
+    tags:
+  pull_request:
+
+jobs:
+  tests:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+
+          - name: Code style checks
+            os: ubuntu-latest
+            python: 3.x
+            toxenv: codestyle
+
+          - name: Python 3.7 with minimal dependencies
+            os: ubuntu-latest
+            python: 3.7
+            toxenv: py37-test
+
+          - name: Python 3.8 with all optional dependencies
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: py38-test-alldeps
+            toxargs: -v --develop
+            toxposargs: --open-files
+
+          # NOTE: In the build below we also check that tests do not open and
+          # leave open any files. This has a performance impact on running the
+          # tests, hence why it is not enabled by default.
+          - name: Python 3.7 with oldest supported version of all dependencies
+            os: ubuntu-16.04
+            python: 3.7
+            toxenv: py37-test-oldestdeps
+
+          - name: Python 3.7 with numpy 1.17 and full coverage
+            os: ubuntu-latest
+            python: 3.7
+            toxenv: py37-test-alldeps-numpy117-cov
+            toxposargs: --remote-data=astropy
+
+          - name: Python 3.8 with all optional dependencies (Windows)
+            os: windows-latest
+            python: 3.8
+            toxenv: py38-test-alldeps
+            toxposargs: --durations=50
+
+          - name: Python 3.7 with all optional dependencies (MacOS X)
+            os: macos-latest
+            python: 3.7
+            toxenv: py37-test-alldeps
+            toxposargs: --durations=50
+
+          - name: Python 3.6 with astropy LTS
+            os: ubuntu-16.04
+            python: 3.7
```

```suggestion
            python: 3.6
```

(I'm not a maintainer so I can't push to your branch)